presto-main/src/main/java/io/prestosql/sql/planner/LogicalPlanner.java

```diff
@@ -340,28 +340,26 @@ private RelationPlan createInsertPlan(Analysis analysis, Insert insertStatement)
 
         TableMetadata tableMetadata = metadata.getTableMetadata(session, insert.getTarget());
 
-        List<ColumnMetadata> visibleTableColumns = tableMetadata.getColumns().stream()
-                .filter(column -> !column.isHidden())
-                .collect(toImmutableList());
-        List<String> visibleTableColumnNames = visibleTableColumns.stream()
-                .map(ColumnMetadata::getName)
-                .collect(toImmutableList());
-
         RelationPlan plan = createRelationPlan(analysis, insertStatement.getQuery());
 
         Map<String, ColumnHandle> columns = metadata.getColumnHandles(session, insert.getTarget());
         Assignments.Builder assignments = Assignments.builder();
+        boolean supportsMissingColumnsOnInsert = metadata.supportsMissingColumnsOnInsert(session, insert.getTarget());
+        ImmutableList.Builder<ColumnHandle> columnsToBeInserted = ImmutableList.builder();
+        ImmutableList.Builder<ColumnMetadata> visibleColumnsBuilder = ImmutableList.builder();
+
         for (ColumnMetadata column : tableMetadata.getColumns()) {
             if (column.isHidden()) {
                 continue;
             }
             Symbol output = symbolAllocator.newSymbol(column.getName(), column.getType());
             int index = insert.getColumns().indexOf(columns.get(column.getName()));
-            if (index < 0) {
```

Comment:
I would refactor this code `index < 0` condition is very misleading. It should be something like `isColumnMissing`. (as separate commit)

Reply:
One is it has a better naming but it introduces a new variable which is required only for that `if` condition.